### PR TITLE
[AMDGPU] Create dir for amdgpu specific machineverifier tests

### PR DESCRIPTION
**Pull request by AditiRM**
Move the AMDGPU target specific testcases in MachineVerifier separately into new directory.
Reference : https://github.com/llvm/llvm-project/pull/105494#discussion_r1735055750

### CODE REVIEW

**Comment by jayfoad** [comment]
Please make the `amdgpu` directory name upper case, like everywhere else in `test/`.

Also it would be good to remove `REQUIRES: amdgpu-registered-target` from each individual test and into a `lit.local.cfg` but I guess that could be done in a follow up.

**Comment by AditiRM** [comment]
@jayfoad Please review the latest changes as per suggestions.

File: llvm/test/MachineVerifier/AMDGPU/lit.local.cfg [hunk 1]
```diff
@@ -0,0 +1,2 @@
+if not "AMDGPU" in config.root.targets:
+    config.unsupported = True
```
Comment (vikramRH) [inline llvm/test/MachineVerifier/AMDGPU/lit.local.cfg:2]: add a newline

Comment (AditiRM) [reply llvm/test/MachineVerifier/AMDGPU/lit.local.cfg]: done.

**Review by cdevadas — APPROVED** [review]

**Review by jayfoad — APPROVED** [review]
LGTM, thanks!

**Pull request merged**